### PR TITLE
Update dependencies and fix compiler errors with recent DMD (v2.109.9)

### PR DESCRIPTION
**Pull request by stonemaster**

### CODE REVIEW

**Review by aminya — APPROVED** [review]

**Review by aminya — CHANGES_REQUESTED** [review]
Could you update the CI as well?

**Comment by stonemaster** [comment]
Sure, I am happy to check what's going on with the CI!

What is your opinion on the compiler versions used: should the old versions be replaced by recent LDC/DMD versions? At a first glance there are some random looking build errors but I don't know whether they are related to old compiler versions ..

**Comment by aminya** [comment]
Yes, I am fine dropping support for old compilers as long as we bump the major version

**Comment by aminya** [comment]
Sub flags are deprecated for dmd, they should be removed from the dub file

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -13,18 +13,18 @@ jobs:
...
-          - macos-11.0
+          - macos-13.0
         d:
-          - "ldc-1.28.1"
-          - "dmd-2.098.1"
+          - "ldc-1.38.0"
+          - "dmd-2.109.1"
         clang:
-          - 13.0.0
+          - 18.0.0
```
Comment (aminya) [inline .github/workflows/CI.yml]:
```suggestion
           # default setup-cpp version 
          - true
```

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -13,18 +13,19 @@ jobs:
       fail-fast: false
       matrix:
         os:
-          - ubuntu-20.04
+          - ubuntu-24.04
```
Comment (aminya) [inline .github/workflows/CI.yml]:
```suggestion
          - ubuntu-22.04
```

**Comment by stonemaster** [comment]
Thanks for the last commits. It seems that two build configurations failed because of "out of memory" :-| 

Locally the `lint` command runs fine for me but outputs the following strange error:
```
...d-tree-sitter/script/lib/meson.d(0:0)[warn]: Variable  is never modified and could have been declared const or immutable.
```

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -13,18 +13,19 @@ jobs:
       fail-fast: false
       matrix:
         os:
-          - ubuntu-20.04
+          - ubuntu-22.04
           - windows-2022
-          - macos-11.0
+          - macos-13.0
```
Comment (aminya) [inline .github/workflows/CI.yml]:
```suggestion
          - macos-13
```

**Review by aminya — APPROVED** [review]

**Pull request merged**